MainWindow.cpp

```diff
@@ -231,18 +236,8 @@ void MainWindow::openProject(std::unique_ptr<buffers::Project> openedProject) {
   treeModel.reset(new TreeModel(project->mutable_game()->mutable_root(), nullptr));
 
   ui->treeView->setModel(treeModel.get());
-  treeModel->connect(treeModel.get(), &QAbstractItemModel::dataChanged,
-                     [=](const QModelIndex &topLeft, const QModelIndex &bottomRight) {
-                       for (int row = topLeft.row(); row <= bottomRight.row(); ++row) {
-                         for (int column = topLeft.column(); column <= bottomRight.column(); ++column) {
-                           auto index = topLeft.sibling(row, column);
-                           buffers::TreeNode *item = static_cast<buffers::TreeNode *>(index.internalPointer());
-                           if (!subWindows.contains(item)) return;
-                           auto subWindow = subWindows[item];
-                           subWindow->setWindowTitle(QString::fromStdString(item->name()));
-                         }
-                       }
-                     });
+  treeModel->connect(treeModel.get(), &TreeModel::ResourceRenamed, resourceMap.get(),
+                     &ResourceModelMap::ResourceRenamed);

```

Comment:
Thank you. Please try to avoid ten-line lambdas. RGM's use of them has become... somewhat excessive.

Reply:
Right, I don't know, we've just sort of used them where it's easier. I've perhaps been abusing them a bit. Having looked this over again, I may later change this completely around again later. One reason to change it again is so the editor subwindow also has the same icon as the tree node.